assets/js/blocks/cart-checkout/cart/frontend.js

```diff
@@ -2,31 +2,44 @@
  * External dependencies
  */
 import { withRestApiHydration } from '@woocommerce/block-hocs';
+import { useCollection } from '@woocommerce/base-hooks';
+import { RawHTML } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import FullCart from './full-cart';
 import renderFrontend from '../../../utils/render-frontend.js';
 
-const isCartEmpty = false; // @todo check if the cart has some products
-const selector = '.wp-block-woocommerce-cart';
-
-if ( ! isCartEmpty ) {
-	const getProps = () => {
-		return {
-			attributes: {},
-		};
-	};
+/**
+ * Wrapper component to supply API data and show empty cart view as needed.
+ */
+const CartFrontend = ( { emptyCart } ) => {
+	const { results: cartData, isLoading } = useCollection( {
+		namespace: '/wc/store',
+		resourceName: 'cart',
+	} );
```

Comment:
I wonder if it makes sense to extract this to a separate file as it's done with [`useStoreProducts`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/hooks/use-store-products.js) or [`useShippingRates`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/hooks/use-shipping-rates.js). This way we separate fetching the data (done in the hook) and displaying it (done in this component file).

Reply:
Good idea, we'd be able to use the hook in the checkout block as well.